src/components/lexical/plugins/ToolbarPlugin/components/LinkEditor.tsx

```diff
@@ -27,33 +31,54 @@ type LinkEditorProps = {
 }
 
 const LinkEditor = ({ isOpen, onOpen, onClose, insertLink }: LinkEditorProps) => {
+  const methods = useForm<{ link: string }>({
+    mode: "all",
+  })
+
+  const {
+    control,
+    setValue,
+    register,
+    formState: { errors },
+  } = methods
+  const link = useWatch({ name: "link", control: control })
+
   const [editor] = useLexicalComposerContext()
 
   const initialFocusRef = useRef<HTMLInputElement>()
-  const [linkUrl, setLinkUrl] = useState("")
   const [lastSelection, setLastSelection] = useState(null)
   const [shouldOpenEditor, setShouldOpenEditor] = useState(false)
 
+  const checkLinkValid = (linkToCheck: string) => {
```

Comment:
Since this function don't have any dependencies, just a simple param, you could move it outside of the component